README.md

```diff
@@ -1,2 +1,116 @@
+
+### Appending messages to disk
+
+```swift
+try myCache.append(message: aMessageInstance)
+```
+
+By the time the above method exits, the message will have been persisted to disk. A CacheAdvance keeps no in-memory buffer. Appending a new message is cheap, as a CacheAdvance needs to encode and persist only the new message and associated metadata.
+
+A CacheAdvance instance that does not overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if appending a message would exceed the cache's `maximumBytes`. A CacheAdvance instance that does overwrite old messages will throw a `CacheAdvanceError.messageDataTooLarge` if the message would require more than `maximumBytes` to store even after evicting all older messages from the cache.
+
+To ensure that caches can be read from 32bit devices, messages should not be larger than 2GB in size.
+
+### Retrieving messages from disk
+
+```swift
+let cachedMessages = try myCache.messages()
+```
+
+This method reads all cached messages from disk into memory.
+
+### Thread safety
+
+CacheAdvances are not thread safe: a single CacheAdvance instance should always be interacted with from a single, serial queue. Since CacheAdvance reads from and writes to the disk synchronously, it is best to interact with a CacheAdvance on a background queue to prevent blocking the main queue.
+
+### Error handling
+
+A CacheAdvance will never fatal error: only recoverable errors will be thrown. A CacheAdvance may throw a `CacheAdvanceError`, or errors related to reading or writing with `FileHandle`s.
+
+If a `CacheAdvanceError.fileCorrupted` error is thrown, the cache file is corrupt and should be deleted.
+
+## How it works
+
+CacheAdvance immediately persists each appended messages to disk using `FileHandle`s. Messages are encoded using a `JSONEncoder`. Messages are written to disk as an encoded data blob that is prefixed with the length of the message. The length of a message is stored using a `UInt32` to ensure that the size of the data on disk that stores a message's length is consistent between devices.
+
+The first 64bytes of a CacheAdvance is reserved for storing metadata about the file. Any configuration data that must be static between cache opens should be stored in this header. It is also reasonable to store mutable information in the header, if doing so speeds up reads or writes to the file. The header format is managed by [FileHeader.swift](Sources/CacheAdvance/FileHeader.swift).
+
+## Requirements
+
+* Xcode 11.2 or later.
+* iOS 12 or later.
+* tvOS 12 or later.
+* watchOS 5 or later.
+* macOS 10.14 or later.
+* Swift 5.0 or later.
+
+## Installation
+
+### Swift Package Manager
+
+To install Relativity in your iOS project with [Swift Package Manager](https://github.com/apple/swift-package-manager), the following lines can be added to your `Package.swift` file:
```

Comment:
Cache Advance not Relativity.

Reply:
😂 I gotta get better at copy/paste. Thank you for catching!